### PR TITLE
Add reserved keywords

### LINKED ISSUE #231: rpc_spec should include a list of the reserved keywords
rpc_spec module should include a list of the reserved keywords for all platforms (iOS, Java Suite, JavaScript suite) so when RPC generators, in the mentioned platforms, try to generate RPCs, they avoid using the reserved keywords.

### PR DESCRIPTION
**Pull request by bilal-alsharifi**
Fixes #231 

This PR adds a list of keywords that RPC Generators in all platforms (iOS, Java Suite, JavaScript Suite) should avoid using by following these rules:
- If the generator needs to create a variable or method and the name was one of the names in the reserved keyword, the generator should append the suffix `Param` to it. For example `getFunctionID`  > `getFunctionIDParam`
- Any empty line in the reserved keywords file should be ignored
- Any line in the reserved keywords file that starts with one or more `#` symbols then `space` should be considered a comment and ignored (like `#<space>` or `##<space>` or `###<space>` …). Note that some reserved keywords have a `#` without a `<space>` so it can't just check for `#`.
- Reserved keywords matching should be case insensitive
- Although the reserved keywords list has a section for each platform, The generator for each platform should avoid using any of the keywords whether it belongs to that platform or another one.

### CODE REVIEW

**Comment by crokita** [comment]
There is a weird case for CancelInteraction where both getFunctionID and setFunctionID are used, but only the getFunctionID is a reserved keyword. If we want consistency with the method name changes, setFunctionID may need to be added to this list so that both will have the same suffix once generated. Thoughts?

**Comment by bilal-alsharifi** [comment]
Good point @crokita I am okay with adding `setFunctionID` to prevent that weird case from happening. What do you think @joeljfischer ?

**Comment by joeljfischer** [comment]
> the generator should append the suffix `param` to it. 

For clarity, I think that should be `Param`.

> Any line in the reserved keywords file that starts with one or more # symbols then space should be considered a comment and ignored

I would add that some reserved keywords have a `#` without a `<space>` so it can't just check for `#`

File: RpcParser/RESERVED_KEYWORDS [hunk 1]
```diff
@@ -0,0 +1,260 @@
+# SDl Java Suite
```
Comment (joeljfischer) [inline RpcParser/RESERVED_KEYWORDS]:
```suggestion
# Java Suite Library
```

File: RpcParser/RESERVED_KEYWORDS [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+setCorrelationID
+getSuccess
+setSuccess
+getResultCode
+setResultCode
+getInfo
+setInfo
+preprocessMsg
+
+# SDL JavaScript Suite
```
Comment (joeljfischer) [inline RpcParser/RESERVED_KEYWORDS]:
```suggestion
# JavaScript Suite Library
```

File: RpcParser/RESERVED_KEYWORDS [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+KEY_SUCCESS
+KEY_INFO
+KEY_RESULT_CODE
+getValue 
+setValue 
+getFunctionID 
+getResultCode 
+setResultCode 
+
+# SDL iOS
```
Comment (joeljfischer) [inline RpcParser/RESERVED_KEYWORDS]:
```suggestion
# iOS Library
```

File: RpcParser/RESERVED_KEYWORDS [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+#fileLiteral
+#function
+#if
+#imageLiteral
+#line
+#selector
+#sourceLocation
+#warning
+
+## SDL Reserved
```
Comment (joeljfischer) [inline RpcParser/RESERVED_KEYWORDS]:
```suggestion
# SDL Reserved
```

**Review by joeljfischer — CHANGES_REQUESTED** [review]

**Comment by joeljfischer** [comment]
I think this should also be case insensitive

**Comment by bilal-alsharifi** [comment]
@joeljfischer I applied your requested changes.

**Review by joeljfischer — APPROVED** [review]

**Comment by crokita** [comment]
Please resolve my previous question about adding setFunctionID to the list before continuing

**Comment by bilal-alsharifi** [comment]
@crokita done.

**Review by crokita — APPROVED** [review]

**Pull request merged**